src/alias/azext_alias/argument.py

```diff
@@ -0,0 +1,165 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=import-error
+
+import re
+import shlex
+
+from knack.util import CLIError
+
+from jinja2 import Template
+from azext_alias._const import (
+    POS_ARG_NAME_REGEX,
+    NUMBER_PLACEHOLDER_REGEX,
+    DUPLICATED_PLACEHOLDER_ERROR,
+    RENDER_TEMPLATE_ERROR,
+    INSUFFICIENT_POS_ARG_ERROR,
+    PLACEHOLDER_EVAL_ERROR,
+    EXPRESSION_REGEX
+)
+
+
+def get_pos_args_names(arg):
+    """
+    Get all the positional arguments' names in order.
+
+    Args:
+        arg: The word which this function performs searching on.
+
+    Returns:
+        A list of positional arguments' names.
+    """
+    pos_args_name = list(map(str.strip, re.findall(POS_ARG_NAME_REGEX, arg)))
+
+    # Make sure there is no duplicated positional argument
+    if len(pos_args_name) != len(set(pos_args_name)):
+        raise CLIError(DUPLICATED_PLACEHOLDER_ERROR.format(arg))
+
+    for i, pos_arg_name in enumerate(pos_args_name):
+        if pos_arg_name.isdigit():
+            pos_args_name[i] = '_{}'.format(pos_arg_name)
+
+    return pos_args_name
+
+
+def stringify_placeholder_expr(arg):
+    """
+    Jinja does not accept numbers as placeholder names,
+    so add a "_" before the number to make them valid placeholder names.
+    Surround placeholders expressions with "" so we can preserve spaces inside the positional arguments.
+
+    Args:
+        arg: The string to process.
+
+    Returns:
+        A processed string where placeholders are surrounded by "" and
+        numbered placeholders are prepended with "_".
+    """
+    number_placeholders = list(map(str.strip, re.findall(NUMBER_PLACEHOLDER_REGEX, arg)))
+    for number_placeholder in number_placeholders:
+        number = re.search(r'\d+', number_placeholder).group()
+        arg = arg.replace(number_placeholder, '{{_' + number + '}}')
```

Comment:
Between the previous method and this, there are two places where the underscore is added to the prefix of a number to avoid the invalid name. I suggest it should be extracted to a method called `normalize_variable_name` or such.